tests/queries/0_stateless/01731_async_task_queue_wait.sh

```diff
@@ -7,4 +7,4 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 # regression for 'Empty task was returned from async task queue' during query
 # cancellation with async_socket_for_remote=1 (that ignores
 # max_distributed_connections)
-$(timeout --signal=SIGINT 1 ${CLICKHOUSE_CLIENT} --max_distributed_connections=1 --max_block_size=2  --interactive_delay=900000 -q "select number + sleep(0.3) as x from remote('127.{2,3}', system.numbers) settings max_block_size = 2") 2>&1 | grep "Empty task was returned from async task queue" || true
+timeout --signal=SIGINT 1 ${CLICKHOUSE_CLIENT} --max_distributed_connections=1 --max_block_size=2  --interactive_delay=900000 -q "select number + sleep(0.3) as x from remote('127.{2,3}', system.numbers) settings max_block_size = 2" 2>&1 | grep "Empty task was returned from async task queue" || true
```

Comment:
without `$()` test does not fail for me

Reply:
But what those brackets does? They should just hide the stdout (the syntax looks odd)
FWIW I've verified this example and it does fails, but not always (like your original)

Maybe you forgot to adjust `listen_host`? (since the error about Connection refused to 127.2/3 error will be hidden now)

```
./clickhouse-debug server -- --listen_host=::
```

Reply:
If tests fails for you, then it is ok.

I thought that `timeout` command may brake `2>&1` redirection, but I seem to be wrong.